pkg/os/shell/shell.go

```diff
@@ -0,0 +1,146 @@
+package shell
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/code-ready/machine/libmachine/shell"
+)
+
+var (
+	supportedShell = []string{"bash", "fish", "powershell", "cmd", "emacs", "tcsh", "zsh"}
```

Comment:
do we really want to support 'fish' again?

Reply:
I would also place this in an `_os.go` file, as powershell is windows only, bash is on both linux and macos... but we do not want to support hthis on the windows bash implementation, right? (we had issues with this before). same for `cmd`, not available on anything else.